### PR TITLE
Chore add workflows snippets for 6.0 docs

### PR DESCRIPTION
**Pull request by jonniebigodes**
This pull request adds the code snippets for the workflows section.

What was done:
- Added the snippets (react and react ts). I know that there's a lot of them, but i was aiming for consistency.
- The markdown files are now referencing the files.

Feel free to provide feedback

### CODE REVIEW

**Review by domyen — APPROVED** [review]

File: docs/snippets/react/component-story-with-query.js.mdx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+```js
+// my-component-with-query.stories.js
+
+import MyComponentThatHasAQuery, { MyQuery} from '../component-that-has-a-query';
+
+export const LoggedOut = () => <MyComponentThatHasAQuery />;
```
Comment (tmeasday) [inline docs/snippets/react/component-story-with-query.js.mdx:6]: Let's use a template.

File: docs/snippets/react/list-story-template.ts.mdx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+```js
+// List.stories.js
+
+import React from 'react';
+import { List, ListProps } from './List';
+import { ListItem, ListItemProps} from './ListItem';
+import { Unchecked } from './ListItem.stories';
+
+const ListTemplate = ({ items, ...args }) => (
```
Comment (tmeasday) [inline docs/snippets/react/list-story-template.ts.mdx:9]: Do we need types here?

**Review by tmeasday — APPROVED** [review]
Looks good, I wonder about the TS snippets. I suppose depending on your TS settings you may or may not have to type your stories--is this right @shilman / @jonniebigodes?

What do we want to do in our snippets?

**Comment by shilman** [comment]
@tmeasday @jonniebigodes i don't really understand the different levels of type strictness in TS. i think given that a lot of TS people like their types strict, we should do our best to provide strictly typed TS snippets. people who want to be loose are welcome to omit whatever their settings allow them to.

**Comment by jonniebigodes** [comment]
@shilman exactly why i did it here. Also as a added bonus, involuntarily we're adding good practices.

File: docs/snippets/react/component-story-with-query.js.mdx (hunk 1 above)
Comment (jonniebigodes) [reply docs/snippets/react/component-story-with-query.js.mdx]: Changed the snippet to include a template with the latest commit.

Let me know if you need anything else.

**Comment by jonniebigodes** [comment]
conflicts are solved, let me know if you need anything else.

**Pull request merged**